src/Microsoft.IdentityModel.Tokens/StandardConfiguration.cs

```diff
@@ -0,0 +1,51 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    ///  Represents a generic metadata configuration which is applicable for both XML and JSON based configurations.
+    /// </summary>
+    public abstract class StandardConfiguration
```

Comment:
Open to feedback on what the names of these new classes should be, as well as whether they should be interfaces or abstract classes.

Reply:
Not sure on what the future holds for this class. Right now seems like an interface makes more sense to my eye. Standard, to me implies some form of implementation (but even just any kind of class implies that which doesn't seem to be the case yet). If it's meant to eventually provide a standard/default implementation this seems okay naming to me, otherwise if this is just defining the contract then that's a better spot for interfaces.